src/provider.js

```diff
@@ -28,8 +28,16 @@ export const availableProviders = [new Java_maven(), new Java_gradle_groovy(), n
 export function match(manifest, providers) {
 	let manifestPath = path.parse(manifest)
 	let provider = providers.find(prov => prov.isSupported(manifestPath.base))
-	if (provider) {
-		return provider
+	if (!provider) {
+		throw new Error(`${manifestPath.base} is not supported`)
 	}
-	throw new Error(`${manifestPath.base} is not supported`)
+
+	if (provider.requireLockFile()) {
```

Comment:
here you can just require a `validateLockFile` where default implementations will just return `undefined` or `null` and other implementations can return a specific error message like:
> `package.json` requires a lock file. Use your package manager to initialize it. Example: npm install

Reply:
@ruromero thanks, I added some commits. Now I am using `validateLockFile` but it does not return anything but it is a `void` function. And in case the lock file is not supported then a throw error i generated by the provider itself, this way the provider can customize the Error message according to the technology it handles. WDYT?